src/coreclr/tests/src/CLRTest.CrossGen.targets

```diff
@@ -124,24 +153,55 @@ if defined RunCrossGen (
 REM CrossGen2 Script
 if defined RunCrossGen2 (
     if defined LargeVersionBubble ( set OptionalArguments=!OptionalArguments! /largeversionbubble)
-    if not exist "$(MSBuildProjectName).org" (
-        call :TakeLock
-        set CrossGen2Status=0
-        if not exist "$(MSBuildProjectName).org" (
-            mkdir IL
-            copy $(MSBuildProjectName).dll IL\$(MSBuildProjectName).dll
-            ren $(MSBuildProjectName).dll $(MSBuildProjectName).org
-            set __Command=!_DebuggerFullPath! "!CORE_ROOT!\CoreRun.exe" "!CORE_ROOT!\crossgen2\crossgen2.dll" %21scriptPath%21$(MSBuildProjectName).org -o:%21scriptPath%21$(MSBuildProjectName).dll --targetarch:x64 -O --inputbubble %21ExtraCrossGen2Args%21 -r:"!CORE_ROOT!\System.*.dll" -r:"!CORE_ROOT!\Microsoft.*.dll" -r:"!CORE_ROOT!\netstandard.dll" -r:"!CORE_ROOT!\mscorlib.dll" -r:%25cd%25\*.dll
-            echo "!__Command!"
-            call !__Command!
-            set CrossGen2Status=!ERRORLEVEL!
-        )
-        call :ReleaseLock
-        IF NOT !CrossGen2Status!==0 (
-            ECHO Crossgen2 failed with exitcode - !CrossGen2Status!
-            Exit /b 1
+    call :TakeLock
+    set CrossGen2Status=0
+    if not exist "IL" (
+        mkdir IL
+        if defined CompositeBuildMode (
+          copy *.dll IL\
+        ) else (
+          copy $(MSBuildProjectName).dll IL\
         )
     )
+
+    set __ResponseFile=!scriptPath!\composite-r2r.dll.rsp
+    del /Q !__ResponseFile!
+
+    set __Command=!_DebuggerFullPath!
+    set __Command=!__Command! "!CORE_ROOT!\CoreRun.exe"
+    set __Command=!__Command! "!CORE_ROOT!\crossgen2\crossgen2.dll"
+    set __Command=!__Command! @"!__ResponseFile!"
+    set __Command=!__Command! !ExtraCrossGen2Args!
+
+    if defined CompositeBuildMode (
+        echo --composite>>!__ResponseFile!
+        echo -o:!scriptPath!composite-r2r.dll>>!__ResponseFile!
+        rem In composite mode, treat all dll's in the test folder as rooting inputs
+        echo !scriptPath!IL\*.dll>>!__ResponseFile!
+    ) else (
+        echo --inputbubble>>!__ResponseFile!
+        echo -o:!scriptPath!$(MSBuildProjectName).dll>>!__ResponseFile!
+        echo -r:%21scriptPath%21IL\*.dll>>!__ResponseFile!
```

Comment:
Maybe I'm parsing batch wrong, but is this missing the input test assembly in non-composite mode?

Reply:
Ahh, right you are, thanks for spotting, will fix.

Reply:
Fixed in 3rd commit.